cloudinit/cmd/devel/logs.py

```diff
@@ -22,11 +24,46 @@
 CLOUDINIT_RUN_DIR = "/run/cloud-init"
 
 
+class ApportFile(NamedTuple):
+    path: str
+    label: str
+
+
+INSTALLER_APPORT_SENSITIVE_FILES = [
+    ApportFile(
+        "/var/log/installer/autoinstall-user-data", "AutoInstallUserData"
+    ),
+    ApportFile("/autoinstall.yaml", "AutoInstallYAML"),
+    ApportFile("/etc/cloud/cloud.cfg.d/99-installer.cfg", "InstallerCloudCfg"),
+]
+
+INSTALLER_APPORT_FILES = [
+    ApportFile("/var/log/installer/ubuntu_desktop_installer.log", "UdiLog"),
+    ApportFile(
+        "/var/log/installer/subiquity-server-debug.log", "SubiquityLog"
+    ),
+    ApportFile(
+        "/var/log/installer/subiquity-client-debug.log", "SubiquityClientLog"
+    ),
+    ApportFile("/var/log/installer/curtin-install.log", "CurtinLog"),
+    ApportFile(
+        "/var/log/installer/subiquity-curtin-install.cfg", "CurtinConfig"
+    ),
+    ApportFile("/var/log/installer/curtin-error-logs.tar", "CurtinError"),
```

Comment:
There have been some back and forth changes with this file recently. I first split it into multiple tar files (one for each curtin stage) and then eventually we decided to merge them back into a single file.

Sadly, we did not preserve the original filename.
The right path for the subiquity version that was shipped with the kinetic ISO would be `curtin-errors.tar` - which in Subiquity we tag as `CurtinErrors` for apport.

It's probably a good thing to expect either `curtin-errors.tar` or `curtin-error-logs.tar` to be present since the desktop canary (for instance) does not yet use the updated subiquity version.

Reply:
Thanks @ogayot and @dbungert for the reviews here (and the #1797) . I've addressed the remaining curtin review comments and put up the following #1812  to round out @ogayot's comments.